model_compiler/src/model_compiler/compilers/onnx_model_to_tensorrt_model.py

```diff
@@ -72,15 +78,22 @@ def compile_source(source: OnnxModel, config: Config) -> TensorRTModel:
 
             builder_config.add_optimization_profile(optimization_profile)
 
-        if config.int8_calibrator:
-            builder_config.int8_calibrator = config.int8_calibrator
-
-        if config.enable_fp16:
+        if config.data_type == 'FP16':
             builder_config.set_flag(tensorrt.BuilderFlag.FP16)
 
-        if config.enable_int8:
+        if config.data_type == 'INT8':
             builder_config.set_flag(tensorrt.BuilderFlag.INT8)
 
+            if config.calibrator:
```

Comment:
If the dtype is INT8 and the user do not pass calibrator config, you will use the default calibrator, but the model user pass is unkonwn, so the calibrate process must be wrong.